src/persist-client/src/read.rs

```diff
@@ -98,18 +140,57 @@ where
     pub async fn poll_next(
         &mut self,
         timeout: Duration,
-    ) -> Result<ListenEvent<K, V, T, D>, LocationError> {
-        todo!("{:?}", timeout)
+    ) -> Result<Vec<ListenEvent<K, V, T, D>>, LocationError> {
```

Comment:
@petrosagg minor change to the public API here (return type went from one to many events). I assume this sort of thing is not going to be an issue, but I'll keep cc'ing you anyway :)